### PR TITLE
http-client-java, emitter, deprecate description and details

### PR DESCRIPTION
**Pull request by haolingdong-msft**
Fix https://github.com/microsoft/typespec/issues/4394

Code changes: 
1. deprecate `description` and `details`, use `doc` and `summary` instead.
2. add `npm run lint` in `Build-Packages.ps1`. So that deprecation warning will be thrown.

I quickly checked the generation diff. The generation diffs are in cases where the operation uses tsp template, previously, the `description` will get tempalte's doc, but now `doc` or  `summary` will not. **It only impacts impl classes**. I think it is acceptable.
eg., [code](https://github.com/microsoft/typespec/pull/4529/files#diff-71f0d715f644a82637d8db144d6d3a999b65d05a7b44bd1570306e46c2c83beeL532), below highlighted template's doc will disappear now.
```diff
  @get
  listWithEtag is StandardResourceOperations.ResourceList<
    Resource,
    ResponseHeadersTrait<EtagResponseEnvelope>
  >;

  /**
++   * Resource list operation template.
   * @template Resource Resource type.
   * @template Traits Object describing the traits of the operation.
   */
  @Foundations.Private.ensureVerb("ResourceList", "GET")
  ResourceList<
    Resource extends TypeSpec.Reflection.Model,
    Traits extends TypeSpec.Reflection.Model = {}
  > is Foundations.ResourceList<
    Resource,
    TraitProperties<Traits & InterfaceTraits, TraitLocation.Parameters, TraitContext.List>,
    Foundations.CustomPage<Resource, Traits & InterfaceTraits>,
    Traits & InterfaceTraits,
    ErrorResponse
  >;
```

### CODE REVIEW

File: packages/http-client-java/Setup.ps1 [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # re-build http-client-java
 npm install 
 npm run clean && npm run build
+npm run lint
```
Comment (haolingdong-msft) [inline packages/http-client-java/Setup.ps1:4]: Add `npm run lint` here.

File: packages/http-client-java/generator/http-client-generator-test/src/main/java/com/_specs_/azure/core/page/implementation/PageClientImpl.java [hunk 2]
```diff
@@ -814,8 +814,6 @@ public PagedIterable<BinaryData> listWithCustomPageModel(RequestOptions requestO
     }
 
     /**
-     * List with Azure.Core.Page&lt;&gt;.
-     * 
      * Get the next page of items.
      * <p><strong>Response Body Schema</strong></p>
```
Comment (weidongxu-microsoft) [inline packages/http-client-java/generator/http-client-generator-test/src/main/java/com/_specs_/azure/core/page/implementation/PageClientImpl.java:818]: It seems all the changes is on "next" op (that created by us)? E.g. here "listWithPageNextSinglePageAsync".

We are fine to change (or even no doc) as these are internal.
But not sure why this only happen to "next" op (this seems not relates to template in tsp)?

Comment (haolingdong-msft) [reply packages/http-client-java/generator/http-client-generator-test/src/main/java/com/_specs_/azure/core/page/implementation/PageClientImpl.java:818]: Thanks for the catch, yeah, this is another case where previous `description` and `detail` have difference from current `doc` and `summary`. But not related with template.
Previously `List with Azure.Core.Page&lt;&gt;.` exists on `detail`, but now it exists on `summary`. So from code-model But
previous:
image
now:
image

It is  strange that it only impacts next operation, guess it's due to codegen logic, but I have to debug to find the cause.

**Review by weidongxu-microsoft — APPROVED** [review]
LGTM

I kind of remember Srikanta mentioned in scrum that there is a lint elsewhere.

File: packages/http-client-java/Setup.ps1 (hunk 1 above)
Comment (srnagar) [reply packages/http-client-java/Setup.ps1:4]: Please add `npm run lint` after this `npm run build` line in [Build-Packages.ps1](https://github.com/microsoft/typespec/blob/main/packages/http-client-java/eng/scripts/Build-Packages.ps1#L75) script.

File: packages/http-client-java/eng/scripts/Build-Packages.ps1 [hunk 3]
```diff
@@ -74,6 +74,9 @@ try {
 
     Invoke-LoggedCommand "npm run build" -GroupOutput
 
+    # linting
+    Invoke-LoggedCommand "npm run build" -GroupOutput
```
Comment (weidongxu-microsoft) [inline packages/http-client-java/eng/scripts/Build-Packages.ps1:78]: @haolingdong-msft `build`?

Comment (weidongxu-microsoft) [reply packages/http-client-java/eng/scripts/Build-Packages.ps1:78]: also there are some not found by lint

```
C:\github\autorest.java\typespec-extension\src\code-model-builder.ts
  2344:18  warning  'details' is deprecated. Use `doc` and `summary` instead  deprecation/deprecation

C:\github\autorest.java\typespec-extension\src\external-schemas.ts
  285:23  warning  'description' is deprecated. Use `doc` and `summary` instead  deprecation/deprecation
  286:49  warning  'description' is deprecated. Use `doc` and `summary` instead  deprecation/deprecation
  288:23  warning  'details' is deprecated. Use `doc` and `summary` instead      deprecation/deprecation
  289:70  warning  'details' is deprecated. Use `doc` and `summary` instead      deprecation/deprecation
```

Comment (haolingdong-msft) [reply packages/http-client-java/eng/scripts/Build-Packages.ps1:78]: Oops. thanks for the catch! Yeah, autorest.java has lint errors.
I will fix it in another pr.

**Pull request merged**